src/CMakeLists.txt

```diff
@@ -347,8 +347,23 @@ else() # Linux
     RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO "${SM_ROOT_DIR}"
   )
 
+  if(${WITH_CRASH_HANDLER})
+    sm_add_compile_definition("${SM_EXE_NAME}" CRASH_HANDLER)
+    if(LINUX)
+      # used only in the ArchHooks_Unix.cpp file
+      sm_add_compile_definition("${SM_EXE_NAME}" BACKTRACE_METHOD_X86_LINUX)
```

Comment:
Again, I suspect there is a better approach for this. If only ArchHooks_Unix.cpp needs this compile definition, why not use `set_source_files_properties` so that the other files don't use unneeded definitions?